cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/persistence/aws/dynamodb/DynamoDocumentStoreTemplate.java

```diff
@@ -290,16 +290,53 @@ public void delete(final Item item, final PersistenceExceptionHandler<?>... pers
         return totalItems;
     }
 
-    private QuerySpec generateQuerySpec(final AttributeQuery query) {
+    private <T extends Item> QuerySpec generateQuerySpec(final AttributeQuery query, final Class<T> itemClass) {
         final QuerySpec querySpec = new QuerySpec().withHashKey(query.getAttributeName(),
                 query.getCondition().getValues().iterator().next());
+
+        if (CompoundAttributeQuery.class.isAssignableFrom(query.getClass())) {
```

Comment:
Might be a good idea to delegate building of all QuerySpec building to a different class rather than partially building then using a different mechanism to add the extra bit needed. "extra bit needed" calls out for subclassing the builder.